app/models/dataset.rb

```diff
@@ -31,7 +33,11 @@ def initialize(hash)
     @released = hash["released"]
     @licence = hash["licence"]
     @licence_other = hash["licence_other"]
-    @licence_custom = hash["licence_custom"]
+
+    if hash["licence_custom"]
+      licence_custom = hash["licence_custom"].match?(/\[.+\]/) ? YAML.safe_load(hash["licence_custom"]).join('<br>') : hash["licence_custom"]
```

Comment:
It would be more ideomatic Ruby to write this as a memoized method. Something like this:

Assuming we store the value from the hash as `raw_licence_custom`.

```ruby
def licence_custom
  @licence_custom ||= if raw_licence_custom.match?(/\[.+\]/)
                        YAML.safe_load(raw_licence_custom).join("<br />")
                      else
                        raw_licence_custom
                      end
end
```

Reply:
Can we not use `hash["licence_custom"]` directly?

Reply:
We could.

Personally I wouldn't because it would mean storing `hash` as an instance variable just for this method, and it's ever so slightly less efficient to do the hash lookup multiple times in this method.

Reply:
ok, lets try your way